src/utils/build-query-field-info.ts

```diff
@@ -0,0 +1,75 @@
+import { DocumentNode, FieldNode, SelectionNode, print } from 'graphql';
+
+/**
+ * Information about a field from the query AST
+ */
+export interface QueryFieldInfo {
+  /** The actual field name (not the alias) */
+  fieldName: string;
+  /** Serialized arguments like "(namespace: \"$app:test\", key: \"config\")" */
+  arguments: string;
+}
+
+/**
+ * Build a comprehensive map of query field information including aliases and arguments
+ *
+ * This traverses the query AST once and extracts:
+ * - Field aliases (itemList -> items)
+ * - Field arguments (metafield(namespace: "...", key: "..."))
+ *
+ * The resulting map is used by convertFixtureToQuery to generate queries with proper
+ * field arguments and to resolve aliased field names in fixture data.
+ *
+ * @param queryAST - The GraphQL query AST
+ * @returns Map where keys are field paths and values contain field info
+ */
+export function buildQueryFieldInfo(queryAST: DocumentNode): Map<string, QueryFieldInfo> {
+  const fieldInfo = new Map<string, QueryFieldInfo>();
+
+  function serializeArguments(args: readonly any[]): string {
+    if (!args || args.length === 0) {
+      return '';
+    }
+
+    const serializedArgs = args.map(arg => {
+      const name = arg.name.value;
+      const value = print(arg.value);
+      return `${name}: ${value}`;
+    }).join(', ');
+
+    return `(${serializedArgs})`;
+  }
+
+  function traverseSelections(selections: readonly SelectionNode[], path: string = ''): void {
+    for (const selection of selections) {
+      if (selection.kind === 'Field') {
+        const field = selection as FieldNode;
+        const actualFieldName = field.name.value;
+        const aliasName = field.alias?.value || actualFieldName;
```

Comment:
Nit: it's not an alias name if we fallback to `actualFieldName`. I would call it `responseKey` or something like that.